src/modules/hyprland/workspaces.cpp

```diff
@@ -136,13 +163,27 @@ Workspaces::~Workspaces() {
   std::lock_guard<std::mutex> lg(mutex_);
 }
 
-WorkspaceDto WorkspaceDto::parse(const Json::Value &value) {
-  return WorkspaceDto{value["id"].asInt()};
-}
+Workspace::Workspace(const Json::Value &value)
+    : id_(value["id"].asInt()),
+      name_(value["name"].asString()),
+      monitor_(value["monitor"].asString()),  // TODO:allow using monitor desc
+      windows_(value["id"].asInt()) {
+  active_ = true;
+  is_special_ = false;
+
+  if (name_.find("name:") == 0) {
+    name_ = name_.substr(5);
+  } else if (name_.find("special") == 0) {
+    name_ = id_ == -99 ? name_ : name_.substr(13);
+    is_special_ = 1;
+  }
+  // spdlog::info("\nNew workspace:\n\tid:({})\n\tname:({})\n\tmonitor:({})\n\tspecial:({})\n", id_,
+  // name_, monitor_, is_special_ ? "yes" : "no"); //make less noisy
```

Comment:
I think it should be eather removed or changed to debug.

Reply:
used it for debugging at start, probably will remove

Reply:
removed